### PR TITLE
chore: create components and folder

### PR DESCRIPTION
**Pull request by SarthakC**
Adds Components 
- atoms/DateOfBirthPicker
- molecules/EmailAndPassInput
- molecules/SignUpDetails
- molecules/TripCard
- organisms/CoverImage
- organisms/Header
- organisms/LoginModal
- organisms/TimeLine
- organisms/TripList
- pages/HomePage
- pages/TimeLinePage

#5

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -5,7 +5,9 @@
   "dependencies": {
     "react": "^16.9.0",
     "react-dom": "^16.9.0",
-    "react-scripts": "3.1.1"
+    "react-scripts": "3.1.1",
+    "semantic-ui-css": "^2.4.1",
```
Comment (nimish-gupta) [inline client/package.json]: why semantic ui?

File: client/src/components/App.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react';
+import Navbar from './Navbar/Navbar';
+
+export default class App extends Component {
+  render() {
+    return (
+      <div>
```
Comment (nimish-gupta) [inline client/src/components/App.jsx]: why not functional component?

File: client/src/components/Navbar/Navbar.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React, { Component } from 'react';
+import { Menu, Button, Icon } from 'semantic-ui-react';
+
+export default class Navbar extends Component {
+  state = {
+    companyName: 'Tripen',
+  };
+
+  render() {
```
Comment (nimish-gupta) [inline client/src/components/Navbar/Navbar.jsx]: Why stateful components?

File: client/src/components/Navbar/Navbar.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      <Menu>
+        <Menu.Item header>{this.state.companyName}</Menu.Item>
+        <Menu.Menu position="right">
+          <Menu.Item>
+            <Button primary>
+              <Icon name="plus"></Icon>
+              Create Timeline
+            </Button>
+          </Menu.Item>
+
```
Comment (nimish-gupta) [inline client/src/components/Navbar/Navbar.jsx]: Why new line?

File: client/package.json (hunk 1 above)
Comment (SarthakC) [reply client/package.json]: Because it has [Placeholders](https://semantic-ui.com/elements/placeholder.html). Which could be useful for an app like ours.
However, we just found out that the modals are not accessible. Hence reconsidering this decision.

File: client/src/components/App.jsx (hunk 2 above)
Comment (SarthakC) [reply client/src/components/App.jsx]: fixed.

File: client/src/components/Navbar/Navbar.jsx (hunk 3 above)
Comment (SarthakC) [reply client/src/components/Navbar/Navbar.jsx]: Was experimenting with boilerplate code. Will amend this right away.

File: client/src/components/Navbar/Navbar.jsx (hunk 4 above)
Comment (SarthakC) [reply client/src/components/Navbar/Navbar.jsx]: This is a part of my Prettier settings. Shall I change it?

File: client/src/components/LoginModal/LoginModal.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import React, { Fragment } from 'react';
+import { Button, Modal } from 'react-bootstrap';
+
+const LoginModal = () => {
+  const [show, setShow] = useState(false);
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  return (
+    <Fragment>
```
Comment (rishabh1403) [inline client/src/components/LoginModal/LoginModal.jsx]: https://reactjs.org/docs/fragments.html#short-syntax

Comment (SarthakC) [reply client/src/components/LoginModal/LoginModal.jsx]: This, somehow hasn't been working for me. In this setup. Maybe I am doing something wrong. I have pushed this for refactoring in the end.

File: client/src/components/App.test.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import App from './App';
+
+describe('<App />', () => {
+  const wrapper = mount(<App />);
+  it('should render', () => {
+    wrapper.debug();
+  });
+  it('should contain the navbar and there should be only one navbar', () => {
```
Comment (nimish-gupta) [inline client/src/components/App.test.js]: try to add new line after every block.

File: client/src/components/molecules/SignUpDetails/SignUpDetails.jsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { Form, Col, FormLabel } from 'react-bootstrap';
+import DateOfBirthPicker from '../../atoms/DateOfBirthPicker/DateOfBirthPicker';
+
+export default function SignUpDetails() {
```
Comment (nimish-gupta) [inline client/src/components/molecules/SignUpDetails/SignUpDetails.jsx]: This can be move to separate PR

File: client/src/components/molecules/SignUpDetails/SignUpDetails.jsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  return (
+    <div>
```
Comment (nimish-gupta) [inline client/src/components/molecules/SignUpDetails/SignUpDetails.jsx]: Why div?

File: client/src/components/molecules/TripCard/TripCard.jsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Card } from 'react-bootstrap';
+
+const TripCard = () => {
+  return (
+    <>
```
Comment (nimish-gupta) [inline client/src/components/molecules/TripCard/TripCard.jsx]: Why fragments?

File: client/src/components/organisms/LoginModal/LoginModal.jsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import { Button, Modal, Form } from 'react-bootstrap';
+import EmailAndPassInput from '../../molecules/EmailAndPassInput/EmailAndPassInput';
+import SignUpDetails from '../../molecules/SignUpDetails/SignUpDetails';
+
+const LoginModal = () => {
```
Comment (nimish-gupta) [inline client/src/components/organisms/LoginModal/LoginModal.jsx]: This should be move to separate PR.

File: client/src/components/organisms/TripList/TripList.jsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Container, Row, Col } from 'react-bootstrap';
+
+import TripCard from '../../molecules/TripCard/TripCard';
+
+export default function TripList() {
```
Comment (nimish-gupta) [inline client/src/components/organisms/TripList/TripList.jsx]: This can be move to another PR.

File: client/src/components/atoms/DateOfBirthPicker/DateOfBirthPicker.jsx [hunk 12]
```diff
@@ -0,0 +1,20 @@
+import React, { useState } from 'react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+const DateOfBirthPicker = () => {
+  const [startDate, setStartDate] = useState(null);
+  return (
+    <DatePicker
+      selected={startDate}
```
Comment (nimish-gupta) [inline client/src/components/atoms/DateOfBirthPicker/DateOfBirthPicker.jsx]: startDate null ?

File: client/src/components/atoms/DateOfBirthPicker/DateOfBirthPicker.jsx [hunk 13]
```diff
@@ -0,0 +1,20 @@
+import React, { useState } from 'react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+const DateOfBirthPicker = () => {
+  const [startDate, setStartDate] = useState(null);
```
Comment (nimish-gupta) [inline client/src/components/atoms/DateOfBirthPicker/DateOfBirthPicker.jsx]: you can set this to current date

File: client/src/components/atoms/DateOfBirthPicker/DateOfBirthPicker.jsx [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+  return (
+    <DatePicker
+      selected={startDate}
+      placeholderText="mm/dd/yyyy"
+      showYearDropdown
```
Comment (nimish-gupta) [inline client/src/components/atoms/DateOfBirthPicker/DateOfBirthPicker.jsx]: move the boolean props in the beginning.

File: client/src/components/molecules/EmailAndPassInput/EmailAndPassInput.jsx [hunk 15]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Form } from 'react-bootstrap';
+
+export default function EmailAndPassInput() {
```
Comment (nimish-gupta) [inline client/src/components/molecules/EmailAndPassInput/EmailAndPassInput.jsx]: why not arrow functions?

File: client/src/components/App.test.js (hunk 6 above)
Comment (SarthakC) [reply client/src/components/App.test.js]: Sure thing!

File: client/src/components/organisms/LoginModal/LoginModal.jsx (hunk 10 above)
Comment (SarthakC) [reply client/src/components/organisms/LoginModal/LoginModal.jsx]: Sure. I would need guidance on that.

File: client/src/components/atoms/DateOfBirthPicker/DateOfBirthPicker.jsx (hunk 12 above)
Comment (SarthakC) [reply client/src/components/atoms/DateOfBirthPicker/DateOfBirthPicker.jsx]: If I put the current date here. It displays instead of the placeholder.

File: client/src/components/molecules/EmailAndPassInput/EmailAndPassInput.jsx [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+        <Form.Control type="email" placeholder="Enter email" />
+        <Form.Text className="text-muted">
+          We'll never share your email with anyone else.
+        </Form.Text>
+      </Form.Group>
+      <Form.Group controlId="formBasicPassword">
+        <Form.Label hidden>Password</Form.Label>
+        <Form.Control type="password" placeholder="Password" />
+      </Form.Group>
+    </>
```
[the remark this answers is not in the document]
Comment (SarthakC) [reply client/src/components/molecules/EmailAndPassInput/EmailAndPassInput.jsx]: React throws an error without fragments.

File: client/src/components/organisms/Header/Header.jsx [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+        <Navbar.Toggle aria-controls="basic-navbar-nav" />
+        <Navbar.Collapse id="basic-navbar-nav">
+          <Nav className="ml-auto">
+            <Button variant="primary">Create Timeline</Button>
+
+            <LoginModal></LoginModal>
+          </Nav>
+        </Navbar.Collapse>
+      </Navbar>
+    </Fragment>
```
[the remark this answers is not in the document]
Comment (SarthakC) [reply client/src/components/organisms/Header/Header.jsx]: Amended.

File: client/src/components/organisms/Header/Header.test.js [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+import Header from './Header';
+
+describe('<Header />', () => {
+  const wrapper = mount(<Header />);
+  describe('Logo', () => {
+    const logoWrapper = wrapper.find('.navbar-brand');
+    it('should contain exactly one logo', () => {
+      expect(logoWrapper.length).toBe(1);
+    });
+    it("logo should say 'Tripen'", () => {
```
[the remark this answers is not in the document]
Comment (SarthakC) [reply client/src/components/organisms/Header/Header.test.js]: Haha yes, I realised. Removing it.

File: client/src/components/organisms/LoginModal/LoginModal.jsx [hunk 19]
```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import { Button, Modal, Form } from 'react-bootstrap';
+import EmailAndPassInput from '../../molecules/EmailAndPassInput/EmailAndPassInput';
+import SignUpDetails from '../../molecules/SignUpDetails/SignUpDetails';
```
[the remark this answers is not in the document]
Comment (SarthakC) [reply client/src/components/organisms/LoginModal/LoginModal.jsx]: This nomenclature is based on Atomic design. I agree with the issue though. We can have a discussion on Slack regarding this.

File: client/src/components/molecules/EmailAndPassInput/EmailAndPassInput.jsx (hunk 16 above)
[the remark this answers is not in the document]
Comment (karansapolia) [reply client/src/components/molecules/EmailAndPassInput/EmailAndPassInput.jsx]: Ah! there are two <Form.Groups />, alright :slightly_smiling_face:

File: client/src/index.js [hunk 20]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import './index.css';
-import App from './App';
+import App from './components/App';
+import 'bootstrap/dist/css/bootstrap.min.css';
```
Comment (p-racoon) [inline client/src/index.js]: Can we avoid this by using CDN directly in public/index.html

File: client/.babelrc [hunk 21]
```diff
@@ -0,0 +1,7 @@
+{
+    "env": {
+        "test": {
+            "plugins": ["@babel/plugin-transform-modules-commonjs", "transform-es2015-modules-commonjs"]
```
Comment (nimish-gupta) [inline client/.babelrc]: why?

File: client/package.json [hunk 22]
```diff
@@ -27,5 +37,16 @@
...
     ]
+  },
+  "devDependencies": {
+    "@babel/plugin-transform-modules-commonjs": "^7.6.0",
+    "babel-plugin-syntax-class-properties": "^6.13.0",
+    "babel-plugin-transform-es2015-modules-commonjs": "^6.26.2",
+    "enzyme": "^3.10.0",
+    "enzyme-adapter-react-16": "^1.14.0"
+  },
+  "moduleNameMapper": {
```
Comment (nimish-gupta) [inline client/package.json]: why?

File: client/src/index.js (hunk 20 above)
Comment (SarthakC) [reply client/src/index.js]: We can, but this helps me import bootstrap classes into my react app. (For intellisense/autocomplete)

File: client/.babelrc (hunk 21 above)
Comment (SarthakC) [reply client/.babelrc]: I downloaded 'react-datepicker' package. And it wasn't compatible with Jest. So I had to transform the module.

File: client/package.json (hunk 22 above)
Comment (SarthakC) [reply client/package.json]: For compatibility with Jest.

File: client/src/components/organisms/TimeLine/TimeLine.jsx [hunk 23]
```diff
@@ -0,0 +1,149 @@
...
+  return (
+    <Container>
+      <ul className="cbp_tmtimeline">
+        <li>
+          <time className="cbp_tmtime" dateTime="2013-04-10 18:30">
+            <span>4/10/13</span> <span>18:30</span>
+          </time>
+          {/* <div className="cbp_tmicon cbp_tmicon-phone" /> */}
+          <IcoMoon icon="phone"></IcoMoon>
+          <div className="cbp_tmlabel">
```
Comment (thapasya-m) [inline client/src/components/organisms/TimeLine/TimeLine.jsx]: classname convention: which one is followed?
a) class_name
b)className
c)class-name
found few variations across loginmodal jsx, test files

Comment (SarthakC) [reply client/src/components/organisms/TimeLine/TimeLine.jsx]: The class names are different because they all come from different sources.

File: client/src/components/organisms/TimeLine/TimeLine.jsx [hunk 24]
```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { Container } from 'react-bootstrap';
+import './TimeLine.css';
```
Comment (karansapolia) [inline client/src/components/organisms/TimeLine/TimeLine.jsx]: So right now, instead of css-in-js we are using css files for each component?

File: client/src/components/organisms/TimeLine/TimeLine.jsx [hunk 25]
```diff
@@ -0,0 +1,149 @@
...
+        <li>
+          <time className="cbp_tmtime" dateTime="2013-04-10 18:30">
+            <span>4/10/13</span> <span>18:30</span>
+          </time>
+          {/* <div className="cbp_tmicon cbp_tmicon-phone" /> */}
+          <IcoMoon icon="phone"></IcoMoon>
+          <div className="cbp_tmlabel">
+            <h2>Ricebean black-eyed pea</h2>
+            <p>
+              Winter purslane courgette pumpkin quandong komatsuna fennel green
```
Comment (karansapolia) [inline client/src/components/organisms/TimeLine/TimeLine.jsx]: All these hard-coded values will be replaced by the axios fetching in some other PR?

File: client/yarn-error.log [hunk 26]
```diff
@@ -0,0 +1,10646 @@
+Arguments:
```
Comment (karansapolia) [inline client/yarn-error.log]: yarn-error.log file should not be pushed to master. add it in .gitignore if it is not there already.

File: client/src/components/App.jsx [hunk 27]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import Header from './organisms/Header/Header';
+import HomePage from './pages/HomePage/HomePage';
+import TimeLinePage from './pages/TimeLinePage/TimeLinePage';
+import { BrowserRouter, Route } from 'react-router-dom';
+
+export default function App() {
+  return (
+    <div>
```
Comment (rishabh1403) [inline client/src/components/App.jsx]: fragment?

File: client/src/components/molecules/EmailAndPassInput/EmailAndPassInput.jsx [hunk 28]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Form } from 'react-bootstrap';
+
+const EmailAndPassInput = () => {
+  return (
+    <>
```
Comment (rishabh1403) [inline client/src/components/molecules/EmailAndPassInput/EmailAndPassInput.jsx]: why fragment?

File: client/src/components/molecules/SignUpDetails/SignUpDetails.jsx [hunk 29]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { Form, Col } from 'react-bootstrap';
+import DateOfBirthPicker from '../../atoms/DateOfBirthPicker/DateOfBirthPicker';
+
+export default function SignUpDetails() {
+  return (
+    <>
```
Comment (rishabh1403) [inline client/src/components/molecules/SignUpDetails/SignUpDetails.jsx]: why fragment?

File: client/src/components/organisms/Header/Header.jsx [hunk 30]
```diff
@@ -0,0 +1,20 @@
...
+
+const Header = () => {
+  return (
+    <Navbar bg="light" expand="lg">
+      <Navbar.Brand href="/">Tripen</Navbar.Brand>
+      <Navbar.Toggle aria-controls="basic-navbar-nav" />
+      <Navbar.Collapse id="basic-navbar-nav">
+        <Nav className="ml-auto">
+          <Button variant="primary">Create Timeline</Button>
+
```
Comment (rishabh1403) [inline client/src/components/organisms/Header/Header.jsx]: space?

File: client/src/components/organisms/Header/Header.test.js [hunk 31]
```diff
@@ -0,0 +1,33 @@
...
+      expect(logoWrapper.length).toBe(1);
+    });
+  });
+  describe('Buttons', () => {
+    const buttonsWrapper = wrapper.find('.btn');
+    it('should contain 2 buttons', () => {
+      expect(buttonsWrapper.length).toBe(2);
+    });
+    it('should contain 1 "Create Timeline" button', () => {
+      expect(buttonsWrapper.contains('Create Timeline')).toBe(true);
```
Comment (rishabh1403) [inline client/src/components/organisms/Header/Header.test.js]: not a good way to test for the presence of button. 
Someonce can change the text anytime. 
Try using data-id etc attribute

File: client/src/components/organisms/LoginModal/LoginModal.jsx [hunk 32]
```diff
@@ -0,0 +1,60 @@
...
+const LoginModal = () => {
+  const [show, setShow] = useState(false);
+  const [signUp, setSignUp] = useState(false);
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+  const initSignUp = () => setSignUp(true);
+  const initSignIn = () => setSignUp(false);
+
+  const action = signUp ? 'Sign-up' : 'Sign In';
+  const buttonMessage = signUp
```
Comment (rishabh1403) [inline client/src/components/organisms/LoginModal/LoginModal.jsx]: give some line breaks. Make it more readbale

File: client/src/components/molecules/EmailAndPassInput/EmailAndPassInput.jsx (hunk 28 above)
Comment (SarthakC) [reply client/src/components/molecules/EmailAndPassInput/EmailAndPassInput.jsx]: The form would not render without the fragment

File: client/src/components/molecules/SignUpDetails/SignUpDetails.jsx (hunk 29 above)
Comment (SarthakC) [reply client/src/components/molecules/SignUpDetails/SignUpDetails.jsx]: The form would not render without the fragment
